jupyterhub/user.py

```diff
@@ -161,9 +161,9 @@ def escaped_name(self):
     @property
     def proxy_path(self):
         if self.settings.get('subdomain_host'):
-            return url_path_join('/' + self.domain, self.server.base_url)
+            return url_path_join('/' + self.domain, self.base_url)
         else:
-            return self.server.base_url
+            return self.base_url
```

Comment:
Just for clarification since it's in the code in various places: when would one use `self.server.base_url` vs `self.base_url`?